here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/SourceIdHandler.java

```diff
@@ -60,70 +60,77 @@ public SourceIdHandler(
-    feature.getProperties().getXyzNamespace().removeTagsWithPrefix(TAG_PREFIX);
+  private void updateTagsWithSourceIdProperty(XyzProperties properties, String sourceId) {
+    properties.getXyzNamespace().removeTagsWithPrefix(TAG_PREFIX);
+    properties.getXyzNamespace().addTag(TAG_PREFIX + sourceId, false);
   }
 
   private Optional<String> getSourceIdFromFeature(XyzProperties properties) {
     try {
-      Map<String, Object> momMetaNs = (Map<String, Object>) properties.get(NS_COM_HERE_MOM_META);
-      Object sourceId = momMetaNs.get(SOURCE_ID);
-      return Optional.ofNullable(sourceId).map(Object::toString);
+      return Optional.ofNullable(properties.get(XyzProperties.HERE_META_NS))
+          .map(Map.class::cast)
+          .map(metaProperties -> metaProperties.get(SOURCE_ID))
+          .map(Object::toString);
     } catch (ClassCastException exception) {
       return Optional.empty();
     }
   }
 
-  public static Optional<POp> transformPopWithSourceId(POp sourcePop) {
-    if (OpType.NOT.equals(sourcePop.op())
-        && sourcePop.children() != null
-        && !sourcePop.children().isEmpty()) {
+  public static Optional<POp> mapIntoTagOperation(POp propertyOperation) {
 
-      POp nestedPropertyOperation = sourcePop.children().get(0);
-      return mapIntoTagOperation(nestedPropertyOperation).map(POp::not);
+    if (sourceIdTransformationCapable(propertyOperation) && operationTypeAllowed(propertyOperation)) {
+      return Optional.of(POp.exists(PRef.tag(TAG_PREFIX + propertyOperation.getValue())));
     }
 
-    return mapIntoTagOperation(sourcePop);
+    return Optional.empty();
   }
 
   private static boolean propertyReferenceEqualsSourceId(PRef pRef) {
     List<@NotNull String> path = pRef.getPath();
-    return path.size() == 3 && path.containsAll(List.of("properties", NS_COM_HERE_MOM_META, SOURCE_ID));
+    return path.size() == PREF_PATHS_SIZE
+        && path.containsAll(List.of(XyzFeature.PROPERTIES, XyzProperties.HERE_META_NS, SOURCE_ID));
   }
 
-  private static Optional<POp> mapIntoTagOperation(POp propertyOperation) {
-
-    if (propertyReferenceEqualsSourceId(propertyOperation.getPropertyRef())
+  private static boolean sourceIdTransformationCapable(POp propertyOperation) {
+    return propertyReferenceEqualsSourceId(propertyOperation.getPropertyRef())
         && propertyOperation.getValue() != null
-        && propertyOperation.children() == null) {
-
-      if (propertyOperation.op().equals(POpType.EQ)
-          || propertyOperation.op().equals(POpType.CONTAINS)) {
-
-        return Optional.of(POp.exists(PRef.tag(TAG_PREFIX + propertyOperation.getValue())));
-      }
-    }
+        && propertyOperation.children() == null;
+  }
 
-    return Optional.empty();
+  private static boolean operationTypeAllowed(POp propertyOperation) {
+    return propertyOperation.op().equals(POpType.EQ)
+        || propertyOperation.op().equals(POpType.CONTAINS);
```

Comment:
nth: you could also have static `Set<POpType> ALLOWED_PROPERTY_OPERATIONS` and this method would simply run `contains` against it. I think there's not too much difference thought, just thinking out loudly